pkg/volume/glusterfs/glusterfs.go

```diff
@@ -811,10 +834,11 @@ func (p *glusterfsVolumeProvisioner) CreateVolume(gid int) (r *v1.GlusterfsVolum
 		return nil, 0, "", fmt.Errorf("failed to create endpoint/service %v/%v: %v", epNamespace, epServiceName, err)
 	}
 	glog.V(3).Infof("dynamic endpoint %v and service %v ", endpoint, service)
-	return &v1.GlusterfsVolumeSource{
-		EndpointsName: endpoint.Name,
-		Path:          volume.Name,
-		ReadOnly:      false,
+	return &v1.GlusterfsPersistentVolumeSource{
+		EndpointsName:      endpoint.Name,
+		EndpointsNamespace: &epNamespace,
```

Comment:
do you plan to adjust the provisioner to allow specifying the endpoint namespace as an option, so it doesn't have to be exposed to the PVC user?

Reply:
As of now, getting the option from storage class is not planned, I can avoid exposing it to pvc user while bringing the change. Is it fine?

Reply:
follow up is fine, though it would be ideal to give the option in the same release the PV capability is added to avoid confusion about availability

Reply:
@humblec external provisioner too?